### PR TITLE
Workaround for 'request' not defined error.

### PR DESCRIPTION
**Pull request by daynew**
When this change was [previously deployed](https://github.com/code-dot-org/code-dot-org/blob/4ee90f6b46be6a8945a0decc579f0db4d8009744/dashboard/lib/cpa.rb#L28), we got the [error](https://app.honeybadger.io/projects/3240/faults/97207022) `NameError: undefined local variable or method 'request' for Cpa:Module` on line 28 of `cpa.rb`:
```
def self.cpa_experience(
    request,
    schedule = experiment_value('cpa_schedule', request), # line 28
```
We don't know why the server thinks the request is defined, so we are going to attempt to workaround it by removing `schedule` and `override` from the function parameters. These were optional parameters only used in unit tests.

[Previous PR](https://github.com/code-dot-org/code-dot-org/pull/52489)

### CODE REVIEW

**Comment by daynew** [comment]
Look at https://github.com/code-dot-org/code-dot-org/pull/52506/commits/71de437a1394f1d37e0205ccaf71b35616c44374 if you want to see the work around specific code

**Review by wilkie — APPROVED** [review]
This tackles my instinct of what the issue is. Regardless, if this doesn't work, this will certainly reveal the problem. LGTM!

I'm crossing my fingers. Thanks for your patience and hard work!

File: dashboard/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -323,6 +326,28 @@ def clear_sign_up_session_vars
     end
   end
 
+  # Check that the user is compliant with the Child Account Policy. If they
+  # are not compliant, then we need to send them to the lockout page.
+  def assert_child_account_policy
+    # Check that the child account policy is currently enabled.
+    return unless Cpa.cpa_experience(request)
```
Comment (artem-vavilov) [inline dashboard/app/controllers/application_controller.rb]: You will have the same error. Try to call it from the main context:
```suggestion
    return unless ::Cpa.cpa_experience(request)
```

Comment (daynew) [reply dashboard/app/controllers/application_controller.rb]: sounds good

**Pull request merged**